pippy/PipelineDriver.py

```diff
@@ -1305,12 +1307,31 @@ class ExecutorDescriptor:
         for stage_id, descr in enumerate(executor_descriptors):
             # Assign stages to rank workers in a round-robin fashion
             rank = self.all_ranks[stage_id % self.world_size]
+            if self.device is not None:
+                logging.debug(
+                    f"[root] Moving stage_id = {stage_id} mod to {self.device}"
+                )
+                descr.mod.to(self.device)
+            logging.debug(f"[root] Sending stage_id = {stage_id} mod to worker")
             self.remote_stage_executor_rrefs[descr.name] = (
                 stage_id,
                 self.rank_worker_rrefs[rank]
                 .remote()
                 .create_stage_executor(stage_id=stage_id, mod=descr.mod),
             )
+            if self.device is not None:
+                logging.debug(
+                    f"[root] Waiting stage_id = {stage_id} mod to be confirmed by worker"
+                )
+                while not self.remote_stage_executor_rrefs[descr.name][
+                    1
+                ].confirmed_by_owner():
+                    pass
```

Comment:
Maybe the above wait should apply generally? (regardless of whether device is None or not)